lib/active_reporting/reporting_dimension.rb

```diff
@@ -149,70 +166,75 @@ def validate_against_datetime_hierarchies(hierarchical_label)
       raise InvalidDimensionLabel, "#{hierarchical_label} is not a valid datetime grouping label in #{name}"
     end
 
+    def validate_label_is_datetime
+      return if dimension_fact_model.model.column_for_attribute(@label).type == :datetime
+      raise InvalidDimensionLabel, "'#{@label}' is not a datetime column"
+    end
+
     def validate_against_fact_model_properties(hierarchical_label)
       return if dimension_fact_model.hierarchical_levels.include?(hierarchical_label.to_sym)
       raise InvalidDimensionLabel, "#{hierarchical_label} is not a hierarchical label in #{name}"
     end
 
-    def degenerate_fragment
-      return "#{name}_#{@label}" if datetime?
-      "#{model.quoted_table_name}.#{name}"
-    end
-
-    def degenerate_select_fragment
-      if datetime?
-        if model.connection.adapter_name == "Mysql2"
-          return degenerate_select_fragment_mysql
-        else # Postgress
-          return degenerate_select_fragment_postgress
-        end
+    def datetime_drill_label_fragment(select_statement)
+      if model.connection.adapter_name == "Mysql2"
+        datetime_drill_mysql(select_statement)
+      else # Postgress
+        datetime_drill_postgress(select_statement)
       end
-      "#{model.quoted_table_name}.#{name}"
     end
 
-    def degenerate_select_fragment_postgress
-      "DATE_TRUNC('#{@label}', #{model.quoted_table_name}.#{name}) AS #{name}_#{@label}"
+    def datetime_drill_postgress(select_statement)
```

Comment:
(This would apply to `select_statement` as well)

This isn't really a select statement that is being passed in here, but rather the column name, right? To me, a "select statement" in SQL is the entire thing between `SELECT` and `FROM`, which could be anything. Instead, I recommend an argument name such as `column` or `dimension_column` or something.

Reply:
Done!